clang/lib/3C/PersistentSourceLoc.cpp

```diff
@@ -67,10 +67,16 @@ PersistentSourceLoc PersistentSourceLoc::mkPSL(clang::SourceRange SR,
   if (TFSL.isValid()) {
     const FileEntry *Fe = SM.getFileEntryForID(TFSL.getFileID());
     std::string ToConv = Fn;
+    StringRef DidConv = Fn;
     std::string FeAbsS = "";
-    if (Fe != nullptr)
+    if (Fe != nullptr) {
       ToConv = std::string(Fe->getName());
-    getCanonicalFilePath(ToConv, FeAbsS);
+      DidConv = Fe->tryGetRealPathName();
+    }
+    if (DidConv.empty())
+      getCanonicalFilePath(ToConv, FeAbsS);
+    else
+      FeAbsS = DidConv.str();
```

Comment:
Can you explain what's going on here? The `ToConv` and `DidConv` variables are initialized to the same `Fn` and in any case I'm not too clear about what's happening here and what's different from what we had before.

Reply:
There are two versions of a function to get the "canonical path" for a function. Matt created one recently, and it didn't work with clang's filesystem, so I had to use one that did. Here, I defaulted back to ours in case that helped. I need to get a final design from @mattmccutchen-cci for this. Shall we use clang's entirely, or do we need to augment it some how for our needs?

Reply:
My final proposed design was in [this comment](https://github.com/correctcomputation/checkedc-clang/pull/488#issuecomment-799859671).  In summary: here I think we can use Clang's entirely, while in RewriteUtils, I'd like to use Clang's followed by ours on the result rather than Clang's falling back to ours.

Reply:
This part looks good to me now.  Thanks.